apis/gpio/src/lib.rs

```diff
@@ -56,11 +56,12 @@ impl Pull for PullNone {
 pub struct Gpio<S: Syscalls>(S);
 
 impl<S: Syscalls> Gpio<S> {
-    /// Run a check against the gpio capsule to ensure it is present.
-    ///
-    /// Returns true` if the driver was present. This does not necessarily mean
-    /// that the driver is working, as it may still fail to allocate grant
-    /// memory.
```

Comment:
It might be a good idea to keep the latter half of this comment.

For another day—it's an interesting point more generally. What is the point of an "exists" check? I might think it's to make sure the driver will be available when you want to call it. Following that, there's a case that "exists" syscalls should do grant allocations, but that's a more fundamental design change.